votingsystem/src/general_meeting.rs

```diff
@@ -0,0 +1,262 @@
+    pub meeting: Option<Box<dyn MeetingManager>>,
+}
+
+impl Context {
+    fn storage(&self) -> &dyn SubStorageAccess {
+        self.storage.as_ref().unwrap().as_ref()
+    }
+
+    fn storage_mut(&mut self) -> &mut dyn SubStorageAccess {
+        self.storage.as_mut().unwrap().as_mut()
+    }
+
+    fn meeting(&self) ->&dyn MeetingManager {
+        self.meeting.as_ref().unwrap().as_ref()
+    }
+    fn meeting_mut(&mut self)->&mut dyn MeetingManager {
+        self.meeting.as_mut().unwrap().as_mut()
+    }
+}
+
+impl Service for Context {}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct TxGeneralMeeting{
+    pub end_time: TimeStamp,
+    pub tallying_time: TimeStamp,
+}
+
+impl Action for TxGeneralMeeting{}
+pub type OwnTransaction = crate::common::SignedTransaction<TxGeneralMeeting>;
+
+impl Context {
+    fn excute_tx(&mut self, transaction: &Transaction) -> Result<GeneralMeetingId, ExecuteError> {
+        if transaction.tx_type() != "CreateGeneralMeeting" {
+            return Err(ExecuteError::InvalidMetadata)
+        }
+        let tx: OwnTransaction = serde_cbor::from_slice(&transaction.body()).map_err(|_| ExecuteError::InvalidFormat)?;
+        tx.verify().map_err(|_| ExecuteError::InvalidSign);
+
+        let end_time = tx.tx.action.end_time;
+        let tallying = tx.tx.action.tallying_time;
+        let meeting = GeneralMeeting::new(end_time,tallying);
+        let meeting_id = &meeting.id;
+        let return_meeting_id = self.meeting_mut().create_general_meeting(meeting.clone(), meeting_id.clone()).unwrap();
+        Ok(return_meeting_id)
+    }
+}
+
+impl Stateful for Context {
+    fn set_storage(&mut self, storage: ServiceRef<dyn SubStorageAccess>) {
+        self.storage.replace(storage.unwrap_import().into_remote());
+    }
+}
+
+impl InitGenesis for Context {
+    fn begin_genesis(&mut self) {}
+
+    fn init_genesis(&mut self, config: &[u8]) {
+    }
+
+    fn end_genesis(&mut self) {}
+}
+
+
+impl TxOwner for Context {
+    fn block_opened(&mut self, _: &Header) -> Result<(), HeaderError> {
+        Ok(())
+    }
+
+    fn execute_transaction(&mut self, transaction: &Transaction) -> Result<TransactionOutcome, ()> {
+        if let Err(error) = self.excute_tx(transaction) {
+            match error {
+                ExecuteError::InvalidMetadata => Err(()),
+                ExecuteError::InvalidSign => Err(()),
+                ExecuteError::InvalidFormat => Err(()),
+            }
+        } else {
+            Ok(Default::default())
+        }
+    }
+
+    fn check_transaction(&self, transaction: &Transaction) -> Result<(), coordinator::types::ErrorCode> {
+        let todo_fixthis: coordinator::types::ErrorCode = 3;
+        assert_eq!(transaction.tx_type(), "CreateGeneralMeeting");
+        let tx: OwnTransaction = serde_cbor::from_slice(&transaction.body()).map_err(|_| todo_fixthis)?;
+        tx.verify().map_err(|_| todo_fixthis)?;
+        Ok(())
+    }
```

Comment:
Shouldn't we check the signer?

Reply:
A verified person must be allowed to create a general meeting.